### ISSUE TITLE
Share Issues Categories for sub-projects(11898,5358) for Redmine 3.4

### ISSUE THREAD

**Issue by y503unavailable** [y503unavailable/redmine#14]
Imprement Share Issues Categories for sub-projects(redmine.org 11898,5358) for Redmine 3.4

Same as below tickets

Inheritable issue categories
http://www.redmine.org/issues/5358 

Share Issues Categories for sub-projects
http://www.redmine.org/issues/11898

**Comment by y503unavailable** [comment]
Made branch from unofficial-34
https://github.com/y503unavailable/redmine/tree/inherit-category34

**Comment by y503unavailable** [comment]
I changed all except view (db, model, controller, helper, lib, locale).

The correction was carried out with reference to the following.
I examined the changes of redmine-org issue 465, 11898.
I searched for the current code (3.4) by sharing and applied the code modification related to version-sharing to issue-categories.
I changed the locale for en and ja.

Confirm that the related basic item operation (category, Issue) can be performed correctly.

Starting Redmine itself.
Confirm addition of items of DB itself.

(Check below in single PJ)
Add Category and save.
Add Issue and save.
Edit and save Issue.
Issue list display.
